src/units/attack_type.hpp

```diff
@@ -104,19 +104,24 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 
 	/** Returns the damage per attack of this weapon, considering specials. */
 	double modified_damage() const;
+	/** Return the defense value, considering specials.
+	 * @param cth The chance_to_hit value modified or not by function.
+	 */
+	int modified_chance_to_hit(int cth) const;
 
 	/** Return the special weapon value, considering specials.
 	 * @param abil_list The list of special checked.
 	 * @param base_value The value modified or not by function.
 	 */
 	int composite_value(const unit_ability_list& abil_list, int base_value) const;
 	/** Returns list for weapon like abilities for each ability type. */
-	unit_ability_list get_weapon_ability(const std::string& ability) const;
+	unit_ability_list get_weapon_ability(const std::string& ability, bool inverse_affect = false) const;
 	/**
 	 * @param special the tag name to check for
+	 * @param inverse_affect If true, the effect of apply_to are inversed
 	 * @return list which contains get_weapon_ability and get_specials list for each ability type, with overwritten items removed
 	 */
-	unit_ability_list get_specials_and_abilities(const std::string& special) const;
+	unit_ability_list get_specials_and_abilities(const std::string& special, bool inverse_affect = false) const;
```

Comment:
What is the `inverse_effect` argument needed for?

Reply:
Defense ability work by affect cth of opponent by default, affect must be inversé in this case and i had coose between use tag_name  =="defense" or create a boolean attribute

Reply:
inverse_effect argument removed